crates/bin/prove_block/src/main.rs

```diff
@@ -290,8 +340,24 @@ async fn main() -> Result<(), Box<dyn Error>> {
         }
     }
 
+    // TODO: clean up...
+    // now that we have the class_hash for each contract, fill in the transaction data with the
+    // class_hash. when transactions were first processed above, this information wasn't available.
+    for transaction in transactions.iter_mut() {
+        if let Some(sender_address) = transaction.sender_address {
+            let class_hash = address_to_class_hash
+                .get(&sender_address)
+                .expect("should have a class_hash for each known contract addresses at this point");
+            log::info!("Filling in class_hash {:x} for txn", class_hash);
+            transaction.class_hash = Some(*class_hash);
+        } else {
+            // TODO: are there txn types which wouldn't have a sender address?
+            log::warn!("Found transaction without sender_address");
+        }
+    }
+
     // write facts from proof
-    for proof in storage_proofs.values() {
+    for proof in storage_proofs.values().chain(previous_storage_proofs.values()) {
```

Comment:
Why do we need this?

Reply:
I imagine it's because we need to write facts for all the storage proofs we fetched. Specifically, we need to prove any reads that we did (`block_number - 1`) and any writes that we do (`block_number`).

@odesenfans wrote this, so maybe he has more context...